dali/ft/daftformat.cpp

```diff
@@ -691,6 +691,31 @@ CCsvPartitioner::CCsvPartitioner(const FileFormat & _format) : CInputBasePartiti
     fields.setown(new KeptAtomTable);
 }
 
+void CCsvPartitioner::getRecordStructure(StringBuffer & _recordStructure)
+{
+    if (nullptr == strstr(recordStructure.str(),"END;"))
```

Comment:
I think this should only ever be called if !srcAttr->hasProp("ECL") in commoned up code (see other comments).

And/or is this is check incase called again? If so, would be clearer if said:
```
if (recordStrcuture.length())
    return;
```